graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
```

Comment:
@nakedible Discussed with @dmcgowan and we think this is okay for this PR.

@ncdc We'll clean this up in the future.